src/module-elasticsuite-catalog/Model/ResourceModel/Product/Indexer/Fulltext/Datasource/InventoryData.php

```diff
@@ -84,16 +84,12 @@ public function loadInventoryData($storeId, $productIds)
         $tableName = $this->stockIndexTableProvider->execute($stockId);
 
         $select = $this->getConnection()->select()
-            ->from(['product' => $this->getTable('catalog_product_entity')], [])
-            ->join(
-                ['stock_index' => $tableName],
-                'product.sku = stock_index.' . IndexStructure::SKU,
-                [
-                    'product_id'    => 'product.entity_id',
-                    'stock_status'  => 'stock_index.' . IndexStructure::IS_SALABLE,
-                    'qty'           => 'stock_index.' . IndexStructure::QUANTITY,
-                ]
-            );
+            ->from($tableName, [
```

Comment:
It will only work if you're only using the default stock : 

- the table `inventory_stock_1` is actually a view based on a join of `cataloginventory_stock` and `catalog_product_entity` so it does provide the product_id
- any subsequent stock index table will only have columns `sku`, `quantity` and `is_salable`

Performance loss due to the extra join is negligible if not non-existent if only using the default stock.